go/integration_test/processor/examples_test.go

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) 2024 Fantom Foundation
+//
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file and at fantom.foundation/bsl11.
+//
+// Change Date: 2028-4-16
+//
+// On the date above, in accordance with the Business Source License, use of
+// this software will be governed by the GNU Lesser General Public License v3.
+
+package processor
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/Fantom-foundation/Tosca/go/examples"
+	"github.com/Fantom-foundation/Tosca/go/tosca"
+)
+
+var (
+	processorExamples = []examples.Example{
+		examples.GetIncrementExample(),
+		examples.GetFibExample(),
+		examples.GetSha3Example(),
+		examples.GetArithmeticExample(),
+		examples.GetMemoryExample(),
+		examples.GetJumpdestAnalysisExample(),
+		examples.GetStopAnalysisExample(),
+		examples.GetPush1AnalysisExample(),
+		examples.GetPush32AnalysisExample(),
+	}
+)
+
+func TestProcessor_Examples(t *testing.T) {
+	for _, example := range processorExamples {
+		for processorName, processor := range getProcessors() {
+			for i := 0; i < 10; i++ {
+				t.Run(fmt.Sprintf("%s-%s-%d", example.Name, processorName, i), func(t *testing.T) {
+					want := example.RunReference(i)
+					got, err := RunOnProcessor(example, processor, i)
+					if err != nil {
+						t.Fatalf("error processing contract: %v", err)
+					}
+					if want != got.Result {
+						t.Fatalf("incorrect result, wanted %d, got %d", want, got.Result)
+					}
+				})
+			}
+		}
+	}
+}
+
+func RunOnProcessor(example examples.Example, processor tosca.Processor, argument int) (examples.Result, error) {
```

Comment:
The `RunOn` for the interpreter is located [here](https://github.com/Fantom-foundation/Tosca/blob/3bd91c86e2aaacf4bd136c2aa4d05b3d7d356a47/go/examples/example.go#L52), inside the example package. Maybe the one for the processor should be placed next to it. This would eliminate the need for making example-package internal fields and functions public and allow the examples -- if this would ever be needed -- to be in control of e.g. block parameters.

The `TransactionContext` derived from the `Scenario` could be passed as an argument to the `RunOnProcessor` function.

Wdyt?